Orange/evaluation/scoring.py

```diff
@@ -11,26 +11,67 @@
 """
 
 import math
-import sys
 
 import numpy as np
 import sklearn.metrics as skl_metrics
+
+from Orange.data import DiscreteVariable, ContinuousVariable
 from Orange.misc.wrapper_meta import WrapperMeta
 
 __all__ = ["CA", "Precision", "Recall", "F1", "PrecisionRecallFSupport", "AUC",
            "MSE", "RMSE", "MAE", "R2", "compute_CD", "graph_ranks", "LogLoss"]
 
 
-class Score(metaclass=WrapperMeta):
+class ScoreMetaType(WrapperMeta):
+    """
+    Maintain a registry of non-abstract subclasses and assign the default
+    value of `name`.
+
+    The existing meta class Registry cannot be used since a meta class cannot
+    have multiple inherited __new__ methods."""
+    def __new__(mcs, name, bases, dict_, **kwargs):
+        cls = WrapperMeta.__new__(mcs, name, bases, dict_)
+        # Essentially `if cls is not Score`, except that Score may not exist yet
+        if hasattr(cls, "registry"):
+            if not kwargs.get("abstract"):
+                # Don't use inherited names, look into dict_
+                cls.name = dict_.get("name", name)
+                cls.registry[name] = cls
+        else:
+            cls.registry = {}
+        return cls
+
+    def __init__(cls, *args, **_):
+        WrapperMeta.__init__(cls, *args)
+
+
+# Derived classes may include a keyword argument name with a short readable
+# name of the score, and an attribute long_name with a few word description.
+#
+# class TGOGScore(Score, name="Thingamagog"):
+#    long_name = "Magic Thingamagog score"
```

Comment:
The comment referred to an older, broken code. `name` is now just a class attribute. I removed the comment.